src/snowflake/snowpark/_internal/analyzer/analyzer.py

```diff
@@ -468,7 +472,7 @@ def do_resolve_with_resolved_children(
 
         if isinstance(logical_plan, TableFunctionRelation):
             return self.plan_builder.from_table_function(
-                self.analyze(logical_plan.table_function)
+                self.analyze(logical_plan.table_function), logical_plan
```

Comment:
Can you help me understand why `logical_plan`/`source_plan` is needed for generator? I might be missing something, but it seems to me that only expressions/columns/aliases are added to `generator`, so `logical_plan` seems a bit too broad.

Reply:
we use it to extract `operators` like `seq1` or `random` by `source_plan.table_function.operators`. I am still debating if I should just pass on operators but most functions in `do_resolve_with_resolved_children` take in `logical_plan` so I followed the same pattern